pkg/apis/common/aws.go

```diff
@@ -0,0 +1,97 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+type AWSCommon struct {
+	// Auth is the S3 authentication (accessKey/secretKey) configuration.
+	Region                 string `json:"region,omitempty"`                 // AWS region
+	ProfileCredentialsName string `json:"profileCredentialsName,omitempty"` // Profile name for profile credentials provider
+	SessionToken           string `json:"sessionToken,omitempty"`           // Session token
+	URIEndpointOverride    string `json:"uriEndpointOverride,omitempty"`    // Override endpoint URI
+	OverrideEndpoint       bool   `json:"overrideEndpoint" default:"false"` // Override endpoint flag
+}
+
+type AWSS3 struct {
+	AWSCommon               `json:",inline"` // Embeds AWSCommon to inherit its fields in JSON
+	BucketNameOrArn         string           `json:"bucketNameOrArn,omitempty"`         // S3 Bucket name or ARN
+	DeleteAfterRead         bool             `json:"deleteAfterRead" default:"true"`    // Auto-delete objects after reading
+	MoveAfterRead           bool             `json:"moveAfterRead" default:"false"`     // Move objects after reading
+	DestinationBucket       string           `json:"destinationBucket,omitempty"`       // Destination bucket for moved objects
+	DestinationBucketPrefix string           `json:"destinationBucketPrefix,omitempty"` // Prefix for moved objects
+	DestinationBucketSuffix string           `json:"destinationBucketSuffix,omitempty"` // Suffix for moved objects
+	AutoCreateBucket        bool             `json:"autoCreateBucket" default:"false"`  // Auto-create S3 bucket
+	Prefix                  string           `json:"prefix,omitempty"`                  // S3 bucket prefix for search
+	IgnoreBody              bool             `json:"ignoreBody" default:"false"`        // Ignore object body
+	ForcePathStyle          bool             `json:"forcePathStyle" default:"false"`    // Force path style for bucket access
+	Delay                   int              `json:"delay" default:"500"`               // Delay between polls in milliseconds
+	MaxMessagesPerPoll      int              `json:"maxMessagesPerPoll" default:"10"`   // Max messages to poll per request
+}
+
+type AWSSQS struct {
+	AWSCommon          `json:",inline"` // Embeds AWSCommon to inherit its fields in JSON
+	QueueNameOrArn     string           `json:"queueNameOrArn,omitempty"`              // SQS Queue name or ARN
+	DeleteAfterRead    bool             `json:"deleteAfterRead" default:"true"`        // Auto-delete messages after reading
+	AutoCreateQueue    bool             `json:"autoCreateQueue" default:"false"`       // Auto-create SQS queue
+	AmazonAWSHost      string           `json:"amazonAWSHost" default:"amazonaws.com"` // AWS host
```

Comment:
do we need to name this `AmazonAWSHost` / `amazonAWSHost` since it's clear this is AWS related ?